eval.go

```diff
@@ -821,3 +830,66 @@ func (d *indexData) regexpToMatchTreeRecursive(r *syntax.Regexp, minTextSize int
 	}
 	return &bruteForceMatchTree{}, false, false, nil
 }
+
+// Limits the number of matches on the SearchResult by the given positive limit,
+// returning the remaining limit, if any.
+func (result *SearchResult) LimitMatches(limit int, chunkMatches bool) int {
+	if chunkMatches {
+		for i := range result.Files {
+			limit = limitChunkMatches(&result.Files[i], limit)
+			if limit == 0 {
+				if i != len(result.Files)-1 {
+					result.Files = result.Files[:i+1]
+				}
+				return 0
+			}
+		}
+	} else {
+		for i := range result.Files {
+			limit = limitLineMatches(&result.Files[i], limit)
+			if limit == 0 {
+				if i != len(result.Files)-1 {
+					result.Files = result.Files[:i+1]
+				}
+				return 0
+			}
+		}
+	}
+	return limit
+}
+
+// Limit the number of ChunkMatches in the given FileMatch, returning the
+// remaining limit, if any.
+func limitChunkMatches(file *FileMatch, limit int) int {
```

Comment:
True.

I willfully ignored Content because manipulating it correctly sounded annoying, if not outright challenging, but you're right that I should give it a go.  There may already be utilities for doing this, I have to do a bit of reading.

And I just totally forgot about SymbolInfo.  I will fix that.

Reply:
SymbolInfo is done.  Still need to do Content.

> There may already be utilities for doing this, I have to do a bit of reading.

There are, but not really usable from this far up the call stack, it's all tied to [precomputed line data in contentprovider](https://github.com/sourcegraph/zoekt/blob/f9b3ea5dcf46088cc41a1b8207f977b28014f94a/contentprovider.go#L333).  I don't see much of a choice but to parse Content ourselves.

Reply:
Content should be good now.  The test case is fairly lame but I've done manual testing as well.